Adjust prints in DefaultRectangular to not depend on IO.chpl

mppf opened the pull request:
In my work on arrays, arrays return by value. As a result, the array auto copy calls something equivalent to array assignment. The result of that is that array copy is now sensitive to resolution order. In particular, there are problems with using IO.chpl from within DefaultRectangular if it could be called in the process of performing an array copy. Since the IO calls in this situation are all for debug output, I created a module that does not use IO.chpl to perform the debug output. This resolves the problem on my arrays branch with tests that check for this debug output.

Passed full local testing.
Reviewed by @daviditen - thanks!

daviditen commented on modules/internal/ChapelDebugPrint.chpl:
```diff
@@ -0,0 +1,120 @@
+      if isNumericType(tmp.type) || isBoolType(tmp.type) {
+        str += tmp:string;
+      } else if isRangeType(tmp.type) {
+        str += tmp.lo:string;
```
Unless there's a resolution order type issue preventing this, you could just cast the range to string here.  That would be better for strided, aligned, and unbounded ranges.

daviditen commented on modules/internal/ChapelDebugPrint.chpl:
```diff
@@ -0,0 +1,120 @@
+        str += "..";
+        str += tmp.hi:string;
+      } else if isTupleType(tmp.type) {
+        str += _stringify_index(tmp);
```
You could also cast the tuple to string directly, except it would be a resolution error if any tuple elements don't have a cast to string defined, instead of "?".

daviditen replied:
Actually, I misread something.  I think it would be a resolution error either way.

daviditen commented:
Looks good overall.  I'd prefer to switch the range->string and tuple->string parts to directly cast to string if that doesn't cause problems.

mppf commented:
@daviditen - see the last commit which addresses your feedback. I'm running full local testing again. Also, it still seems to resolve the problems on my arrays branch.

daviditen commented:
@mppf - Looks good.  Just curious, in the scripts where you converted '==' into '=', did you run into a problem with '=='?  Or was that just a "while I'm looking at it anyway..." type of change?

mppf commented:
The == into = change was because I was seeing error messages in coming from the test run. They didn't seem to cause test failures, but I'm sure that they meant the scripts weren't working as intended. == is just not the way to do string comparison inside of an if [ ] in a bash/sh script. (There are plenty of other places where == can do string comparison, just not there).
